src/main/webapp/app/hooks/useOverflowDetector.tsx

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export interface useOverflowDetectorProps {
+  onChange?: (overflow: boolean) => void;
+  handleHeight?: boolean;
```

Comment:
Maybe change to `ignoreWidth` and `ignoreHeight`? Feels more clear to me

Reply:
Yeah good point on the naming and I changed to `detectWidth` and `detectHeight` to denote whether we want to detect x/y overflow. I think its easier to understand positive logic, so will stick to that. Let me know if that's sounds good with you.